x-pack/lib/filters/geoip/download_manager.rb

```diff
@@ -28,75 +27,70 @@ def initialize(database_type, metadata)
   GEOIP_ENDPOINT = "#{GEOIP_HOST}#{GEOIP_PATH}".freeze
 
   public
-  # Check available update and download it. Unzip and validate the file.
-  # return [has_update, new_database_path]
+  # Check available update and download them. Unzip and validate the file.
+  # if the download failed, valid_download return false
+  # return Array of [database_type, valid_download, dirname, new_database_path]
   def fetch_database
-    has_update, database_info = check_update
-
-    if has_update
-      new_database_path = unzip download_database(database_info)
-      assert_database!(new_database_path)
-      return [true, new_database_path]
-    end
-
-    [false, nil]
-  end
-
-  def database_name
-    @database_name ||= "#{DB_PREFIX}#{@database_type}"
-  end
-
-  def database_name_ext
-    @database_name_ext ||= "#{database_name}.#{DB_EXT}"
+    dirname = Time.now.to_i.to_s
+    check_update
+      .map do |database_type, db_info|
+        begin
+          new_zip_path = download_database(database_type, dirname, db_info)
+          new_database_path = unzip(database_type, dirname, new_zip_path)
+          assert_database!(new_database_path)
+          [database_type, true, dirname, new_database_path]
+        rescue => e
+          logger.error(e.message, error_details(e, logger))
+          [database_type, false, nil, nil]
+        end
+      end
   end
   
   private
-  # Call infra endpoint to get md5 of latest database and verify with metadata
-  # return [has_update, server db info]
+  # Call infra endpoint to get md5 of latest databases and verify with metadata
+  # return Array of new database information [database_type, db_info]
   def check_update
     uuid = get_uuid
     res = rest_client.get("#{GEOIP_ENDPOINT}?key=#{uuid}&elastic_geoip_service_tos=agree")
     logger.debug("check update", :endpoint => GEOIP_ENDPOINT, :response => res.status)
 
-    dbs = JSON.parse(res.body)
-    target_db = dbs.select { |db| db['name'].eql?("#{database_name}.#{GZ_EXT}") }.first
-    has_update = @metadata.gz_md5 != target_db['md5_hash']
-    logger.info "new database version detected? #{has_update}"
+    service_resp = JSON.parse(res.body)
 
-    [has_update, target_db]
-  end
+    updated_db = DB_TYPES.map do |database_type|
+      db_info = service_resp.select { |db| db['name'].eql?("#{GEOLITE}#{database_type}.#{GZ_EXT}") }.first
```

Comment:
select + first = find

```suggestion
      db_info = service_resp.find { |db| db['name'].eql?("#{GEOLITE}#{database_type}.#{GZ_EXT}") }
```